internal/controller/servermaintenance_controller.go

```diff
@@ -108,6 +107,16 @@ func (r *ServerMaintenanceReconciler) handlePendingState(ctx context.Context, lo
 			return ctrl.Result{}, nil
 		}
 	}
+	defer func() {
```

Comment:
Instead of using `defer` (I know other projects are using this pattern in their controller impl.) can we put this at the correct place in the reconciliation loop? We are not using `defer` based cleanups/updates in our code atm.

Reply:
ok I just wanted to reduce redundancy.
Removed it